controllers/vspheremachine_controller.go

```diff
@@ -291,19 +291,16 @@ func (r *machineReconciler) reconcileNormal(ctx context.Context, machineCtx capv
 		return reconcile.Result{}, nil
 	}
 
-	//nolint:gocritic
 	if r.supervisorBased {
 		err := r.setVMModifiers(ctx, machineCtx)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-	} else {
+	} else if !machineCtx.GetCluster().Status.InfrastructureReady {
 		// vmwarev1.VSphereCluster doesn't set Cluster.Status.Ready until the API endpoint is available.
-		if !machineCtx.GetCluster().Status.InfrastructureReady {
-			log.Info("Cluster infrastructure is not ready yet")
-			conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
-			return reconcile.Result{}, nil
-		}
+		log.Info("Cluster infrastructure is not ready yet")
+		conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
+		return reconcile.Result{}, nil
```

Comment:
While we're here we should improve the comments and maybe try to simplify this bit of code. Maybe something like:

```go
	// Cluster `.status.infrastructureReady == false is handled differently depending on if the machine is supervisor based.
	// 1) If the Cluster is not supervisor-based mark the VMProvisionedCondition false and return nil.
	// 2) If the Cluster is supervisor-based continue to reconcile as InfrastructureReady is not set to true until after the kube apiserver is available.
	if !machineCtx.GetCluster().Status.InfrastructureReady {
		if !r.supervisorBased {
			log.Info("Cluster infrastructure is not ready yet")
			conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
			return reconcile.Result{}, nil
		}
		// vmwarev1.VSphereCluster doesn't set Cluster.Status.Ready until the API endpoint is available so continue to reconcile.
		// Set modifiers on the supervisor VM.
		err := r.setVMModifiers(ctx, machineCtx)
		if err != nil {
			return reconcile.Result{}, err
		}
	}
```
We should definitely add commments like that - not sure if the code is really better, especially given how we have `if supervisorBased` in other parts of the code.

@chrischdi WDYT?

Reply:
I think for simplicity and readability we should keep the

```go
if r.supervisorBased {
  // supervisor
} else {
  // non supervisor
}
```

pattern. Otherwise it may get more complicated and reduce readability.

Reply:
Hi folks, I think both of your suggestions make sense to me. What about:
1. To set the outer condition, use `if !machineCtx.GetCluster().Status.InfrastructureReady`, and for the inner condition, use `if r.supervisorBased`.
    1. From my understanding, if the cluster is already ready, there may be no need to keep running 'setVMModifiers'.
2. Still using the `r.supervisorBased` structure to ensure readability.
3. Add the necessary comments as mentioned above.

And the code would be something like:
```
	// Cluster `.status.infrastructureReady == false is handled differently depending on if the machine is supervisor based.
	// 1) If the Cluster is not supervisor-based mark the VMProvisionedCondition false and return nil.
	// 2) If the Cluster is supervisor-based continue to reconcile as InfrastructureReady is not set to true until after the kube apiserver is available.
	if !machineCtx.GetCluster().Status.InfrastructureReady {
		if !r.supervisorBased {
			log.Info("Cluster infrastructure is not ready yet")
			conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
			return reconcile.Result{}, nil
		} else {
			// vmwarev1.VSphereCluster doesn't set Cluster.Status.Ready until the API endpoint is available so continue to reconcile.
			// Set modifiers on the supervisor VM.
			err := r.setVMModifiers(ctx, machineCtx)
			if err != nil {
				return reconcile.Result{}, err
			}
		}
	}
```

Reply:
Guys, I have an idea. I reversed the if condition, and the `golangci-lint` check passed. Is this okay with you?
```
	// Cluster `.status.infrastructureReady == false is handled differently depending on if the machine is supervisor based.
	// 1) If the Cluster is not supervisor-based mark the VMProvisionedCondition false and return nil.
	// 2) If the Cluster is supervisor-based continue to reconcile as InfrastructureReady is not set to true until after the kube apiserver is available.
	if !r.supervisorBased {
		// vmwarev1.VSphereCluster doesn't set Cluster.Status.Ready until the API endpoint is available.
		if !machineCtx.GetCluster().Status.InfrastructureReady {
			log.Info("Cluster infrastructure is not ready yet")
			conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
			return reconcile.Result{}, nil
		}
	} else {
		err := r.setVMModifiers(ctx, machineCtx)
		if err != nil {
			return reconcile.Result{}, err
		}
	}
```